src/txins.rs

```diff
@@ -0,0 +1,121 @@
+//! Revault txins
+//! Wrappers around bitcoin's OutPoint and previous TxOut to statically check Revault transaction
+//! creation and ease PSBT management.
+
+use crate::txouts::{CpfpTxOut, FeeBumpTxOut, UnvaultTxOut, VaultTxOut};
+
+use bitcoin::{OutPoint, TxIn};
+
+use std::fmt;
+
+/// A transaction input used by a Revault transaction.
+pub trait RevaultTxIn<T>: fmt::Debug + Clone + PartialEq {
+    /// Get the actual outpoint
+    fn outpoint(&self) -> OutPoint;
+    /// Get a reference to the txout this txin refers
+    fn txout(&self) -> &T;
+    /// Get the actual txout this txin refers
+    fn get_txout(self) -> T;
```

Comment:
rename to `as_outpoint`, `as_txout()`, `into_txout()` as per https://github.com/re-vault/revault/issues/19 ?

Reply:
I considered doing an actual cleanup PR for this and the documentation (maybe some more test cov as well). I could avoid introducing more here but... That incurs some non-trivial rebases (opened the issue after finishing this, fwiw)